### PR TITLE
Update stable nightly branch properly

### PR DESCRIPTION
**Pull request by CoryMartin-NOAA**
This PR changes how the stable-nightly branch is created. It's been running for some time now and I think it is working as intended with the exception of that it will complain and send an email if there are no updates to develop. Which needs to have a good solution (perhaps I don't trap that single exit code?)

### CODE REVIEW

File: ci/stable_driver.sh [hunk 1]
```diff
@@ -73,15 +73,69 @@ $my_dir/stable_mark.sh $stableroot/$datestr/GDASApp
...
+  # commit this change and push
   git add CMakeLists.txt
+  total=$(($total+$?))
+  if [ $total -ne 0 ]; then
+    echo "Unable to add CMakeLists to commit" >> $stableroot/$datestr/output
+  fi
   git commit -m "Update to new stable build on $datestr"
+  total=$(($total+$?))
+  caution=""
+  if [ $total -ne 0 ]; then
```
Comment (CoryMartin-NOAA) [inline ci/stable_driver.sh:128]: This exit code is the troublemaker. Should I just remove lines 114-119? Or do we live with the emails @RussTreadon-NOAA @guillaumevernieres

Comment (CoryMartin-NOAA) [reply ci/stable_driver.sh:128]: Or do we want to echo a date/time to a file to ensure there is always a commit?

Comment (RussTreadon-NOAA) [reply ci/stable_driver.sh:128]: @CoryMartin-NOAA , I'm doing what you probably already did - searching stackoverflow  to see how others handle this.   There are a few hits (e.g., [how-to-git-commit-nothing-without-an-error](https://stackoverflow.com/questions/8123674/how-to-git-commit-nothing-without-an-error)), but I'm not sure if we want to try any.   

We want to capture errors if something goes wrong when we commit real change(s). 
 We don't want to capture _nothing to commit_ as a error.

**Review by guillaumevernieres — DISMISSED** [review]
:+1:

File: ci/stable_driver.sh (hunk 1 above)
Comment (CoryMartin-NOAA) [reply ci/stable_driver.sh:128]: Yep, my current thought is to do something like:
```
date > commit_generated
git add commit_generated
```
to somewhere near lines 108

Comment (guillaumevernieres) [reply ci/stable_driver.sh:128]: I'm fine with the darth-vader email @CoryMartin-NOAA .

**Comment by RussTreadon-NOAA** [comment]
I, too, can live with the no commit email for the time being.   It seems odd to create a commit when there's really nothing to commit.

**Comment by CoryMartin-NOAA** [comment]
@RussTreadon-NOAA so what about if I can have it actually check if the CMakeLists file is updated and if so, not even try to commit and exit early? Perhaps that is the route to go... let me think about how best to do that

**Comment by RussTreadon-NOAA** [comment]
Here's an idea from stackoverflow:  [how-to-git-commit-with-persistent-exit-0-status](https://stackoverflow.com/questions/47036884/how-to-git-commit-with-persistent-exit-0-status).   Key text from this link is pasted below

_So, you're having the classic exited with a 1 problem because you tried to commit when there was nothing to commit. But you still want it to exited with a 1 if there is an actual problem. Just do this: don't run the commit command at all if you find that there is nothing to commit. You will check beforehand if there are any commits to be had, like so:_

`git diff-index --quiet HEAD || git commit -am 'Jolly ol commit message'`

**Comment by CoryMartin-NOAA** [comment]
Thanks @RussTreadon-NOAA I'll try that!

**Comment by CoryMartin-NOAA** [comment]
@RussTreadon-NOAA hopefully now it will run the g-w tests before updating... let's test that tonight before review/merging in this branch.

**Comment by RussTreadon-NOAA** [comment]
Thank you @CoryMartin-NOAA for making this change.   I'm manually cloning & building g-w and GDASApp `develop` on Hera right now.   I'll run `test_gdasapp` once the build completes.

**Comment by RussTreadon-NOAA** [comment]
Clean install of g-w `develop` at [`47f1954b`](https://github.com/NOAA-EMC/global-workflow/commit/47f1954b5c73b2fbb2e38e3a165bf3c2520147d2) and GDASApp `develop` at 2895f79 on Hera yields 48/48 _Passed_ for `test_gdasapp`.
```
(gdasapp) Hera(hfe11):/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/new/sorc/gdas.cd/build$ ctest -R test_gdasapp
Test project /scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/new/sorc/gdas.cd/build
      Start 1433: test_gdasapp_check_python_norms
 1/48 Test #1433: test_gdasapp_check_python_norms .......................   Passed    3.47 sec
      Start 1434: test_gdasapp_check_yaml_keys
 2/48 Test #1434: test_gdasapp_check_yaml_keys ..........................   Passed    0.46 sec
      Start 1435: test_gdasapp_jedi_increment_to_fv3
 3/48 Test #1435: test_gdasapp_jedi_increment_to_fv3 ....................   Passed    8.11 sec
      Start 1436: test_gdasapp_genYAML_prep
 4/48 Test #1436: test_gdasapp_genYAML_prep .............................   Passed    0.03 sec
      Start 1437: test_gdasapp_genYAML_run
 5/48 Test #1437: test_gdasapp_genYAML_run ..............................   Passed    4.07 sec
      Start 1438: test_gdasapp_genYAML_land_prep
 6/48 Test #1438: test_gdasapp_genYAML_land_prep ........................   Passed    0.03 sec
      Start 1439: test_gdasapp_genYAML_land_run
 7/48 Test #1439: test_gdasapp_genYAML_land_run .........................   Passed    1.54 sec
      Start 1440: test_gdasapp_genYAML_aero_prep
 8/48 Test #1440: test_gdasapp_genYAML_aero_prep ........................   Passed    0.03 sec
      Start 1441: test_gdasapp_genYAML_aero_run
 9/48 Test #1441: test_gdasapp_genYAML_aero_run .........................   Passed    1.16 sec
      Start 1442: test_gdasapp_convert_ewok_yaml
10/48 Test #1442: test_gdasapp_convert_ewok_yaml ........................   Passed    0.25 sec
      Start 1443: test_gdasapp_setup_cycled_exp
11/48 Test #1443: test_gdasapp_setup_cycled_exp .........................   Passed    1.00 sec
      Start 1444: test_gdasapp_convert_bufr_temp_dbuoy
12/48 Test #1444: test_gdasapp_convert_bufr_temp_dbuoy ..................   Passed    3.95 sec
      Start 1445: test_gdasapp_convert_bufr_salt_dbuoy
13/48 Test #1445: test_gdasapp_convert_bufr_salt_dbuoy ..................   Passed    0.45 sec
      Start 1446: test_gdasapp_convert_bufr_temp_mbuoyb
14/48 Test #1446: test_gdasapp_convert_bufr_temp_mbuoyb .................   Passed    0.27 sec
      Start 1447: test_gdasapp_convert_bufr_salt_mbuoyb
15/48 Test #1447: test_gdasapp_convert_bufr_salt_mbuoyb .................   Passed    0.21 sec
      Start 1448: test_gdasapp_convert_bufr_tesacprof
16/48 Test #1448: test_gdasapp_convert_bufr_tesacprof ...................   Passed    0.24 sec
      Start 1449: test_gdasapp_convert_bufr_trkobprof
17/48 Test #1449: test_gdasapp_convert_bufr_trkobprof ...................   Passed    0.21 sec
      Start 1450: test_gdasapp_convert_bufr_sfcships
18/48 Test #1450: test_gdasapp_convert_bufr_sfcships ....................   Passed    0.25 sec
      Start 1451: test_gdasapp_convert_bufr_sfcshipsu
19/48 Test #1451: test_gdasapp_convert_bufr_sfcshipsu ...................   Passed    0.33 sec
      Start 1452: test_gdasapp_soca_obsdb
20/48 Test #1452: test_gdasapp_soca_obsdb ...............................   Passed    2.23 sec
      Start 1453: test_gdasapp_soca_nsst_increment_to_mom6
21/48 Test #1453: test_gdasapp_soca_nsst_increment_to_mom6 ..............   Passed   16.99 sec
      Start 1454: test_gdasapp_soca_prep
22/48 Test #1454: test_gdasapp_soca_prep ................................   Passed    3.77 sec
      Start 1455: test_gdasapp_soca_concatioda
23/48 Test #1455: test_gdasapp_soca_concatioda ..........................   Passed    5.50 sec
      Start 1456: test_gdasapp_soca_run_clean
24/48 Test #1456: test_gdasapp_soca_run_clean ...........................   Passed    0.05 sec
      Start 1457: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_PREP
25/48 Test #1457: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_PREP ....   Passed   43.57 sec
      Start 1458: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT
26/48 Test #1458: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT ....   Passed   42.98 sec
      Start 1459: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_RUN
27/48 Test #1459: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_RUN .....   Passed   42.93 sec
      Start 1460: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_CHKPT
28/48 Test #1460: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_CHKPT ...   Passed   42.25 sec
      Start 1461: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_POST
29/48 Test #1461: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_POST ....   Passed   10.25 sec
      Start 1462: test_gdasapp_land_create_ens
30/48 Test #1462: test_gdasapp_land_create_ens ..........................   Passed    1.10 sec
      Start 1463: test_gdasapp_land_imsproc
31/48 Test #1463: test_gdasapp_land_imsproc .............................   Passed    4.02 sec
      Start 1464: test_gdasapp_land_apply_jediincr
32/48 Test #1464: test_gdasapp_land_apply_jediincr ......................   Passed    3.47 sec
      Start 1465: test_gdasapp_land_letkfoi_snowda
33/48 Test #1465: test_gdasapp_land_letkfoi_snowda ......................   Passed   10.52 sec
      Start 1466: test_gdasapp_convert_bufr_adpsfc
34/48 Test #1466: test_gdasapp_convert_bufr_adpsfc ......................   Passed    6.51 sec
      Start 1467: test_gdasapp_convert_gsi_satbias
35/48 Test #1467: test_gdasapp_convert_gsi_satbias ......................   Passed    4.29 sec
      Start 1468: test_gdasapp_store_gsi_satbias
36/48 Test #1468: test_gdasapp_store_gsi_satbias ........................   Passed    1.45 sec
      Start 1469: test_gdasapp_run_jedi_exe_3dhofx
37/48 Test #1469: test_gdasapp_run_jedi_exe_3dhofx ......................   Passed   15.47 sec
      Start 1470: test_gdasapp_run_jedi_exe_3dvar
38/48 Test #1470: test_gdasapp_run_jedi_exe_3dvar .......................   Passed   12.31 sec
      Start 1471: test_gdasapp_run_jedi_exe_3denvar
39/48 Test #1471: test_gdasapp_run_jedi_exe_3denvar .....................   Passed   12.62 sec
      Start 1472: test_gdasapp_run_jedi_exe_letkf
40/48 Test #1472: test_gdasapp_run_jedi_exe_letkf .......................   Passed    9.61 sec
      Start 1473: test_gdasapp_setup_atm_cycled_exp
41/48 Test #1473: test_gdasapp_setup_atm_cycled_exp .....................   Passed    0.59 sec
      Start 1474: test_gdasapp_atm_jjob_var_init
42/48 Test #1474: test_gdasapp_atm_jjob_var_init ........................   Passed   43.38 sec
      Start 1475: test_gdasapp_atm_jjob_var_run
43/48 Test #1475: test_gdasapp_atm_jjob_var_run .........................   Passed   74.11 sec
      Start 1476: test_gdasapp_atm_jjob_var_final
44/48 Test #1476: test_gdasapp_atm_jjob_var_final .......................   Passed   42.33 sec
      Start 1477: test_gdasapp_atm_jjob_ens_init
45/48 Test #1477: test_gdasapp_atm_jjob_ens_init ........................   Passed   43.64 sec
      Start 1478: test_gdasapp_atm_jjob_ens_run
46/48 Test #1478: test_gdasapp_atm_jjob_ens_run .........................   Passed  170.12 sec
      Start 1479: test_gdasapp_atm_jjob_ens_final
47/48 Test #1479: test_gdasapp_atm_jjob_ens_final .......................   Passed   42.15 sec
      Start 1480: test_gdasapp_aero_gen_3dvar_yaml
48/48 Test #1480: test_gdasapp_aero_gen_3dvar_yaml ......................   Passed    0.46 sec

100% tests passed, 0 tests failed out of 48

Total Test time (real) = 736.50 sec
```

**Review by guillaumevernieres — DISMISSED** [review]
Looks good. Thanks @CoryMartin-NOAA .

**Comment by CoryMartin-NOAA** [comment]
Still not 100% sure this is working properly... need to triple check

**Comment by CoryMartin-NOAA** [comment]
Can I get a review of this soon so that we can merge this PR?

**Review by guillaumevernieres — APPROVED** [review]
:+1:

**Pull request merged**